File: qlty-cli/tests/lang/ruby/basic.in/identical.rb
def f0(numbers)
    mean = numbers.sum.to_f / numbers.size

    sorted_numbers = numbers.sort
    median = if sorted_numbers.size % 2 == 0
        mid_index = sorted_numbers.size / 2
        (sorted_numbers[mid_index - 1] + sorted_numbers[mid_index]) / 2
    else
        sorted_numbers[sorted_numbers.size / 2]
    end

    [mean, median]
end

def f1(numbers)
    mean = numbers.sum.to_f / numbers.size

    sorted_numbers = numbers.sort
    median = if sorted_numbers.size % 2 == 0
        mid_index = sorted_numbers.size / 2
        (sorted_numbers[mid_index - 1] + sorted_numbers[mid_index]) / 2
    else
        sorted_numbers[sorted_numbers.size / 2]
    end

    [mean, median]
end

File: qlty-cli/tests/lang/rust/basic.in/identical.rs
fn f0(numbers: &[i32]) -> (f64, Option<i32>) {
    let mean: f64 = numbers.iter().sum::<i32>() as f64 / numbers.len() as f64;

    let mut sorted_numbers = numbers.to_vec();
    sorted_numbers.sort();
    let median: Option<i32> = if sorted_numbers.len() % 2 == 0 {
        let mid_index = sorted_numbers.len() / 2;
        Some((sorted_numbers[mid_index - 1] + sorted_numbers[mid_index]) / 2)
    } else {
        Some(sorted_numbers[sorted_numbers.len() / 2])
    };

    (mean, median)
}

fn f1(numbers: &[i32]) -> (f64, Option<i32>) {
    let mean: f64 = numbers.iter().sum::<i32>() as f64 / numbers.len() as f64;

    let mut sorted_numbers = numbers.to_vec();
    sorted_numbers.sort();
    let median: Option<i32> = if sorted_numbers.len() % 2 == 0 {
        let mid_index = sorted_numbers.len() / 2;
        Some((sorted_numbers[mid_index - 1] + sorted_numbers[mid_index]) / 2)
    } else {
        Some(sorted_numbers[sorted_numbers.len() / 2])
    };

    (mean, median)
}
